adaptor-rm-webapp/src/main/java/com/sample/rm/RMToolManager.java

```diff
@@ -58,34 +63,56 @@ private static int randomNumber(int origin, int bound) {
 		return ThreadLocalRandom.current().nextInt(origin, bound);
 	}
 
-	private static Requirement createRandomRequirement(String id) {
-		Requirement r = null;
-		try {
-			r = RMToolResourcesFactory.createRequirement(id);
-			r.setTitle("A sample Requirement with id:" + id);
-			r.setDescription("A sample Requirement with id:" + id);
-		} catch (URISyntaxException e) {
+    public static ArrayList<Requirement> getRequirements() {
+        return new ArrayList<Requirement>(requirements.values());
```

Comment:
what's this?! we need to change the service to:

- not modify the list (I hope I did that a long time ago, but that only works for `List`s)
- accept any kind of `Iterable` (will require replacing the `List::subList` call I used for the step above)

plus, why are we getting ALL requirements all in one go?